packages/core/custom-branding/core-custom-branding-browser-internal/src/custom_branding_service.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { of, Subject, Observable } from 'rxjs';
+import { shareReplay, takeUntil } from 'rxjs/operators';
+import type {
+  CustomBrandingStart,
+  CustomBrandingSetup,
+  CustomBrandingSetupDeps,
+} from '@kbn/core-custom-branding-browser';
+import type { CustomBranding } from '@kbn/core-custom-branding-common';
+
+export class CustomBrandingService {
+  private customBranding: CustomBranding = {};
+  private hasCustomBranding$: Observable<boolean> = new Observable<boolean>();
+  private customBranding$: Observable<CustomBranding> = new Observable<CustomBranding>();
```

Comment:
NIT/optional: I think we could avoid the `customBranding`+`customBranding$` 'duplication' by having a `BehaviorSubject<CustomBranding>` and using `customBranding$.value` when we need to access it synchronously like in `hasCustomBranding`.

We probably don't even need to old a reference of `hasCustomBranding$` either, and could probably just inline the definition

```ts
return {
   hasCustomBranding$: this.customBranding$.pipe(
       takeUntil(this.stop$), 
       map((customBranding) => Object.keys(customBranding).length > 0)
       shareReplay(1),
   )
}


```